### ISSUE TITLE
Could a --routes-refresh parameter be implemented?

### ISSUE THREAD

**Issue by Codrax** [itzg/mc-router#271]
I would like to run around 10 servers simultaneously, and currently, to add/remove a server I need to change the `config.json`. And the proxy need to be closed and re-opened to index the changes.

With a `--routes-refresh` parameter, optionally taking a second input for how often to refresh the routes, the application could compare the last stored modification date of the file provided with `--routes-config`.

This would allow the server list to be updated gracefully without restarting all servers.

Thank you!

**Comment by itzg** [comment]
That's a great idea!

**Comment by itzg** [comment]
@Codrax with [this library](https://github.com/fsnotify/fsnotify) it looks like I could have it watch for file changes. Would that be even better or still prefer/recommend a timer based approach?

**Comment by Codrax** [comment]
> @Codrax with [this library](https://github.com/fsnotify/fsnotify) it looks like I could have it watch for file changes. Would that be even better or still prefer/recommend a timer based approach?

Both cases would be great as long as the file reloads. I don't know a lot about that library, but I'm pretty sure the timer based approach is simpler as that just utilises the CPUs task scheduler, which I think is more optimised. But both options are valid.

**Comment by zarlo** [comment]
at lest for the fsnotify backend for linux the CPU over head is very little

**Comment by itzg** [comment]
> at lest for the fsnotify backend for linux the CPU over head is very little

Right. It's driven by existing filesystem OS events, so there's virtually no extra overhead.

**Comment by lobo235** [comment]
Was curious if you were able to implement a watch/reload on the config file as discussed. It would be very useful for a problem I'm trying to solve with the way I run my Minecraft servers.

**Comment by itzg** [comment]
Sorry, no, not yet

**Comment by Codrax** [comment]
> Was curious if you were able to implement a watch/reload on the config file as discussed. It would be very useful for a problem I'm trying to solve with the way I run my Minecraft servers. See [#208 (comment)](https://github.com/itzg/mc-router/issues/208#issuecomment-2867965213)

Exactly due to this issue I've developed this alternative software:
https://github.com/Codrax/minecraft-server-map

I had a feeling adding this feature to `mc-router` would take a while.

I'm running my own minecraft server platform, and I needed to add/remove servers without disconnecting all users.

Although, nowadays I use a newer version of my software which can be compiled cross-platform and provides far more features and customization options, I have yet publised that version to GitHub, but could in the near feature If there is demand for such.

**Comment by Codrax** [comment]
I would happily make a pull request myself, but I don't know how to write code in the Go library...

**Comment by itzg** [comment]
Sorry for the long delay in getting around to this one. Release https://github.com/itzg/mc-router/releases/tag/1.30.0 now includes a `--routes-watch` option.
